### PR TITLE
Issue #2829646 by alunyov, rebuilding integration with ip_geoloc and smart_ip

### PR DESCRIPTION
**Pull request by olunov**
Rebuilt integration with location provider. Now instead of loading location data from Telize GeoIP service, it is possible to use HTML5 widget (default), smart_ip or ip_geoloc modules

### CODE REVIEW

File: personalization.admin.inc [hunk 1]
```diff
@@ -125,21 +125,58 @@ function personalization_admin() {
...
+  );
+
+  // @TODO: rebuild loading options with hook for extending.
+  // Check if Smart IP is installed.
+  if (module_exists('smart_ip')) {
+    // Don't translate title of the module.
+    $pz_geo_options['smart_ip'] = 'Smart IP';
+  }
+
+  // Check if Smart IP is installed.
```
Comment (Dan-Ashdown) [inline personalization.admin.inc]: Comment is incorrect

File: personalization.admin.inc [hunk 2]
```diff
@@ -125,21 +125,58 @@ function personalization_admin() {
...
+<ul>
+  <li><strong>Default</strong> - requesting the location from the user will require them to be using a HTML5 compliant browser and your Drupal theme to be HTML5</li>
+  <li><strong>Smart IP</strong> - geo location data are loaded from !smart_ip_link module.</li>
+  <li><strong>IP Geolocation Views & Maps</strong> - geo location data are loaded from !ip_geoloc_link module.</li>
+</ul>
+If you don't see the option make sure corresponding module is installed. Please check README.txt for details.
+", array(
+    '!smart_ip_link' => l('Smart IP', 'https://www.drupal.org/project/smart_ip', array('attributes' => array('target' => '_blank'))),
+    '!ip_geoloc_link' => l('IP Geolocation Views & Maps', 'https://www.drupal.org/project/ip_geoloc', array('attributes' => array('target' => '_blank'))),
+  ));
```
Comment (Dan-Ashdown) [inline personalization.admin.inc]: Would be nicer as a them('item_list' but not a blocker

File: personalization.tracker.inc [hunk 3]
```diff
@@ -327,6 +327,7 @@ function personalization_get_user() {
     }
     else {
       $uid = uniqid(NULL, TRUE);
+      $_COOKIE['pz_user_id'] = $uid;
```
Comment (Dan-Ashdown) [inline personalization.tracker.inc:330]: Not from this PR but I think we have a bit of a security hole with this cookie. We use it's value to write and update data in the database but there's nothing stopping you changing your cookie and affecting someone elses content.

For example if there was a page on your site about my product, I could influence peoples personalisations score by pretending to be them so my content is surfaced.

Comment (olunov) [reply personalization.tracker.inc:330]: I would make it as separate issue on d.o. for easier tracking it.

File: personalization.admin.inc (hunk 1 above)
Comment (olunov) [reply personalization.admin.inc]: Updated

File: personalization.admin.inc (hunk 2 above)
Comment (olunov) [reply personalization.admin.inc]: Done.

**Pull request merged**